api/v1beta1/machineregistration_types.go

```diff
@@ -85,6 +88,14 @@ func (m MachineRegistration) GetClientRegistrationConfig(cacert string) *Config
 
 	mRegistration := conf.Elemental.Registration
 
+	if !meta.IsStatusConditionTrue(m.Status.Conditions, ReadyCondition) {
+		return nil, fmt.Errorf("machine registration is not ready")
+	}
+
+	if m.Status.RegistrationURL == "" {
```

Comment:
probably this check could be omitted and relay on the condition only, however since the bug report is really hard to reproduce I'd say this does not hurt here and helps to prevent errors in case someone manually messed up with machine registration fields.

Reply:
No, I think the check is valid: the URL may be not set if the Rancher installation has not initialized the URL yet (you do once you log in there).
If you create the SeedImage via kubectl before Rancher URL is initialized it will fail creating the URL here.
My guess is that once Rancher URL is initialized, you keep the MachineRegistration there without the URL... creating the SeedImage in that case we end up here.
So, a corner case, but I think this is needed 👍🏼